contracts/GnosisSafePersonalEdition.sol

```diff
@@ -90,29 +157,93 @@ contract GnosisSafePersonalEdition is MasterCopy, GnosisSafe, SignatureValidator
     {
         uint256 startGas = gasleft();
         // We don't provide an error message here, as we use it to return the estimate
+        // solium-disable-next-line error-reason
         require(execute(to, value, data, operation, gasleft()));
         uint256 requiredGas = startGas - gasleft();
         // Convert response to string and return via error message
         revert(string(abi.encodePacked(requiredGas)));
     }
 
-    function checkHash(bytes32 txHash, bytes signatures)
-        internal
-        view
+    /**
+    * @dev Marks a message as signed
+    * @param _data Arbitrary length data that should be marked as signed on the behalf of address(this)
+    */ 
+    function signMessage(bytes _data) 
```

Comment:
This function allows to approve a transaction without collecting signature off-chain. It is valid even if owners are changing. It can only be invalidated, if owners do a delegate-call to invalidate the confirmation. The advantage is questionable as there is no guarantee the confirmation is permanent.